apps/platform/trpc/routes/branch.ts

```diff
@@ -158,4 +157,36 @@ export const branches = createRouter({
 
       return updatedBranch;
     }),
+  delete: withAuth
+    .input(z.object({ branchId: z.string() }))
+    .mutation(async ({ ctx, input }) => {
+      const { branchId } = input;
+
+      const branch = await ctx.prisma.branch.findUnique({
+        where: { id: branchId },
+        select: { protected: true },
+      });
+
+      if (branch?.protected) {
```

Comment:
Does this case handles if the `branch` is not found with the particular associated `branchId` ?

Reply:
Good catch, lets add branch presence validation, and it'll be good to go.

Reply:
Sure, sorry I didn’t see this comment earlier.